core/capabilities/webapi/trigger_test.go

```diff
@@ -0,0 +1,260 @@
+		TriggerID: triggerID2,
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+		Config: Config2,
+	}
+	channel2, err := th.trigger.RegisterTrigger(ctx, triggerReq2)
+	require.NoError(t, err)
+
+	t.Run("happy case single topic to single workflow", func(t *testing.T) {
+		gatewayRequest := gatewayRequest(t, `["daily_price_update"]`)
+
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+		th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
+
+		sent := <-channel
+		require.NotNil(t, sent)
+		require.Equal(t, sent.Event.TriggerType, "web-trigger@1.0.0")
+		// TODO: how to index into Outputs which is wrapped structure and so contains Underlying, thence Params, ie
+		// require.Equal(t, sent.Event.Outputs.Underlying.["Params"]["Topics"], []string{"daily_price_update"})
+	})
+
+	t.Run("happy case single different topic 2 workflows.", func(t *testing.T) {
+		gatewayRequest := gatewayRequest(t, `["ad_hoc_price_update"]`)
+
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+		th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
+
+		sent := <-channel
+		require.NotNil(t, sent)
+		require.Equal(t, sent.Event.TriggerType, "web-trigger@1.0.0")
+		sent2 := <-channel2
+		require.NotNil(t, sent2)
+		require.Equal(t, sent2.Event.TriggerType, "web-trigger@1.0.0")
+	})
+
+	t.Run("happy case empty topic 2 workflows", func(t *testing.T) {
+		gatewayRequest := gatewayRequest(t, `[]`)
+
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+		th.connector.On("SendToGateway", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+		th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
+
+		sent := <-channel
+		require.NotNil(t, sent)
+		require.Equal(t, sent.Event.TriggerType, "web-trigger@1.0.0")
+		sent2 := <-channel2
+		require.NotNil(t, sent2)
+		require.Equal(t, sent2.Event.TriggerType, "web-trigger@1.0.0")
+	})
+
+	t.Run("sad case topic with no workflows", func(t *testing.T) {
+		gatewayRequest := gatewayRequest(t, `["foo"]`)
+
+		th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
+
+	})
+}
+
+func TestRegisterInvalidSender(t *testing.T) {
+	th := setup(t)
+	ctx := testutils.Context(t)
+	Config, _ := workflowTriggerConfig(th, []string{"5"}, []string{"daily_price_update"})
+
+	triggerReq := capabilities.TriggerRegistrationRequest{
+		TriggerID: triggerID1,
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+		Config: Config,
+	}
+	_, err := th.trigger.RegisterTrigger(ctx, triggerReq)
+	require.NoError(t, err)
+
+	gatewayRequest := gatewayRequest(t, `["ad_hoc_price_update"]`)
+
+	th.trigger.HandleGatewayMessage(ctx, "gateway1", gatewayRequest)
```

Comment:
Check that there are no messages on either channel

Reply:
In this case, it won't make sense because the registerTrigger will now fail so we won't have any channels to check.